### PR TITLE
build: update to husky@6

### PR DESCRIPTION
**Pull request by damingerdai**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by gkalpak — APPROVED** [review]
Thx, @damingerdai!
Should we switch to `prepare` npm script (instead of `postinstall`) as is the new recommended way (after typicode/husky#890)?

**Comment by gkalpak** [comment]
BTW, you need to add a commit message body (of at least 20 characters) to pass linting.

**Comment by damingerdai** [comment]
> Should we switch to `prepare` npm script (instead of `postinstall`) as is the new recommended way (after [typicode/husky#890](https://github.com/typicode/husky/pull/890))?

I agree with you, but I am not sure whether I should complete the changes in this pr.

**Comment by damingerdai** [comment]
@gkalpak please review

**Comment by gkalpak** [comment]
@damingerdai, I don't see any changes. Did you forget to push them maybe?

**Comment by damingerdai** [comment]
> @damingerdai, I don't see any changes. Did you forget to push them maybe?

i just add commit message body.

**Comment by gkalpak** [comment]
Oh, I see, sorry 😅
I thought you implemented switching to `prepare` and was looking for code changes 😇

I think it is reasonable to switch to `prepare` as part of this PR, since it is a new recommendation in husky 6, so it makes sense to make the change along with the updating imo.

**Comment by damingerdai** [comment]
@gkalpak please review

File: package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "ts-circular-deps:check": "yarn -s ng-dev ts-circular-deps check --config ./packages/circular-deps-test.conf.js",
     "ts-circular-deps:approve": "yarn -s ng-dev ts-circular-deps approve --config ./packages/circular-deps-test.conf.js",
     "ng-dev": "node dev-infra/ng-dev",
-    "ng-dev:dev": "ts-node --transpile-only -- dev-infra/cli.ts"
+    "ng-dev:dev": "ts-node --transpile-only -- dev-infra/cli.ts",
+    "prepare": "husky install"
```
Comment (gkalpak) [inline package.json:40]: Please move this script just below `postinstall` (since that is conceptually closer to how the scripts are called).

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx for making the changes, @damingerdai :+1:

Can you please change the second commit message to mention typicode/husky#890, where the recommendation to switch to `prepare` is made. For example:

> build: install husky in \`prepare\` script instead of \`postinstall\`
>
> With typicode/husky#890, the recommended way to install husky is in the
> \`prepare\` script instead of the \`postinstall\`. This commit moves
> the husky installation to the \`prepare\` script to align with the new
> recommendation.

**Comment by damingerdai** [comment]
@gkalpak thanks for your suggestions. please review again.

**Review by gkalpak — APPROVED** [review]
Thx again, @damingerdai :+1:

LGTM, except you need to rebase on latest master to make CI happy :smiley:

**Comment by damingerdai** [comment]
@gkalpak i have rebased on latest master but there still are two ci failed

**Review by gkalpak — APPROVED** [review]
Thx again, @damingerdai!
The latest CI failures were flakes. I have restarted the job and it passes now 🚀

**Pull request closed without merging**